alpaca/hypothesis_test.go

```diff
@@ -0,0 +1,259 @@
+package alpaca
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"encoding/base64"1
+	"encoding/json"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"testing/quick"
+
+	"github.com/shopspring/decimal"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// ---------------------------
+// Extended Environment & Auth Tests
+// ---------------------------
+
+// Test that NewClient falls back in the proper order:
+// OAuth > Broker > API keys.
+func TestAuthenticationPriority(t *testing.T) {
+	// When OAuth is provided, it should override all other credentials.
+	client := NewClient(ClientOpts{
+		OAuth:        "oauthtoken",
+		APIKey:       "apikey",
+		APISecret:    "apisecret",
+		BrokerKey:    "brokerkey",
+		BrokerSecret: "brokersecret",
+		BaseURL:      "http://dummy",
+	})
+	req, _ := http.NewRequest(http.MethodGet, client.opts.BaseURL, nil)
+	// In defaultDo, OAuth header should be set:
+	req.Header.Set("Authorization", "Bearer "+client.opts.OAuth)
+	assert.Equal(t, "Bearer oauthtoken", req.Header.Get("Authorization"))
```

Comment:
This makes no sense: it asserts to something that is explicitly set in the test.